### PR TITLE
feat: autostart + single-instance

### PR DESCRIPTION
**Pull request by brayo-pip**
Fixes https://github.com/ActivityWatch/aw-tauri/issues/14 https://github.com/ActivityWatch/aw-tauri/issues/15

> [!IMPORTANT]
> Add autostart and single-instance features using Tauri plugins, updating configuration and documentation.
> 
>   - **Features**:
>     - Add `tauri-plugin-autostart` and `tauri-plugin-single-instance` to `src-tauri/src/lib.rs` for autostart and single-instance handling.
>     - Configure autostart with `MacosLauncher::LaunchAgent` and enable it with flags `--flag1`, `--flag2`.
>     - Implement single-instance logic to quit if another instance is running.
>   - **Configuration**:
>     - Update `package.json` to include `@tauri-apps/plugin-autostart` and `@tauri-apps/cli`.
>     - Update `Cargo.toml` to add `tauri-plugin-single-instance` and `tauri-plugin-autostart` dependencies.
>     - Add `desktop.json` for desktop capabilities and permissions.
>   - **Documentation**:
>     - Update `README.md` to mark autostart and single-instance features as completed.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ActivityWatch%2Faw-tauri&utm_source=github&utm_medium=referral)<sup> for 39f130debc1076cd77010d631430e3993d312bc8. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by brayo-pip** [comment]
#15 #14

**Comment by brayo-pip** [comment]
@ErikBjare review.

**Review by ErikBjare — APPROVED** [review]
Looks great!

File: src-tauri/src/lib.rs [hunk 1]
```diff
@@ -42,9 +44,25 @@ fn greet(name: &str) -> String {
...
+                // Get the autostart manager
+                let autostart_manager = app.autolaunch();
+                // Enable autostart
+                let _ = autostart_manager.enable();
+                // Check enable state
+                println!(
+                    "registered for autostart? {}",
+                    autostart_manager.is_enabled().unwrap()
+                );
+
```
[the remark this answers is not in the document]
Comment (brayo-pip) [reply src-tauri/src/lib.rs:65]: Can be toggled via the frontend UI or maybe a config file. I will modify it to add the toggle via both. We can decide which to keep later.

**Comment by brayo-pip** [comment]
> Feel free to merge :)

Just one more change... This one is a mess.

**Comment by brayo-pip** [comment]
@ErikBjare I do not have merge permissions in this repo

**Comment by brayo-pip** [comment]
> Please merge responsibly (squash messy commits, otherwise rebase, etc.)
Will do 🫡

**Pull request merged**